### PR TITLE
fix: keyboard transition issue when switching from attachment picker to keyboard

### LINKED ISSUE #2024: Keyboard transition broken
**Describe the bug**
There's an unusual transition when going from the media library back to the keyboard as shown in the video posted below. This is for iOS and Android on Simulator and on on a physical device.

**Dev environment info (please complete/provide the following information):**

- Package version (stream-chat-react-native/stream-chat-expo): 5.11.2
- react-native version: 0.70.5
- expo version: 47.0.0
- Device/Emulator (android/ios) and OS version: all

**Expected behavior**
A smooth transition from the media picker to the keyboard

**Screenshots**
https://user-images.githubusercontent.com/35746593/227387306-4a4c416c-05e8-471f-88e5-8e6876903b23.mov

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

Fixes #2024 

Before:

https://github.com/GetStream/stream-chat-react-native/assets/39884168/4215f48d-126c-4a60-ae06-533a64cbbc78

After:

https://github.com/GetStream/stream-chat-react-native/assets/39884168/bb7540a8-a854-4e9e-9d70-c6ed877395cc

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/src/components/AttachmentPicker/AttachmentPicker.tsx [hunk 1]
```diff
@@ -193,7 +199,7 @@ export const AttachmentPicker = React.forwardRef(
         // To keep compatibility with older versions of React Native, where `remove()` is not available
         Keyboard.removeListener(keyboardShowEvent, closePicker);
```
Comment (santhoshvai) [inline package/src/components/AttachmentPicker/AttachmentPicker.tsx:200]:
```suggestion
        Keyboard.removeListener(keyboardShowEvent, onKeyboardOpenHandler);
```

File: package/src/components/AttachmentPicker/AttachmentPicker.tsx [hunk 2]
```diff
@@ -181,8 +181,14 @@ export const AttachmentPicker = React.forwardRef(
     }, [selectedPicker, closePicker]);
 
     useEffect(() => {
+      const onKeyboardOpenHandler = () => {
+        if (selectedPicker) {
+          setSelectedPicker(undefined);
+        }
```
Comment (santhoshvai) [inline package/src/components/AttachmentPicker/AttachmentPicker.tsx:187]: can this be done like this 

setSelectedPicker(undefined)

there is no need for an if at all.. if you want it to set it to undefined whenever keyboard is open

**Review by santhoshvai — CHANGES_REQUESTED** [review]
please see comments

File: package/src/components/AttachmentPicker/AttachmentPicker.tsx (hunk 2 above)
Comment (khushal87) [reply package/src/components/AttachmentPicker/AttachmentPicker.tsx:187]: Ideally, it won't create much of a difference. Its just a sanity check that we set it as undefined only when there is a value in the selectedPicker state. The same is done for the back action handler in the file.

**Pull request merged**